tests_e2e/orchestrator/lib/agent_test_suite.py

```diff
@@ -308,9 +307,9 @@ def execute(self, node: Node, variables: Dict[str, Any], log: Logger) -> None:
                     if not self.context.skip_setup:
                         self._setup_node()
 
-                    test_suites: List[TestSuiteDescription] = AgentTestLoader(self.context.test_source_directory).load(self.context.test_suites)
-
-                    for suite in test_suites:
+                    # pylint seems to think self.context.test_suites is not iterable. Suppressing warning, since its type is List[AgentTestSuite]
+                    #  E1133: Non-iterable value self.context.test_suites is used in an iterating context (not-an-iterable)
+                    for suite in self.context.test_suites:  # pylint: disable=E1133
                         test_suite_success = self._execute_test_suite(suite) and test_suite_success
```

Comment:
how this suite type as TestSuiteInfo. I see `self.context.test_suites` get assigned from the variable we pass to Lisa run. That would be list of commas separated or directory type of single test.

Reply:
The string we pass to the runbook is actually an argument for the combinator:
```
  #
  # These variables parameters for the AgentTestSuitesCombinator combinator
  #
  # The test suites to execute
  - name: test_suites
    value: "agent_bvt"
    is_case_visible: true

```
Then the combinator produces an array of TestSuiteInfos that is passed to the AgentTestSuite in the "test_suites_info" variable:

```
  #
  # These variables are set by the AgentTestSuitesCombinator combinator
  #
...
  - name: test_suites_info
    value: []
    is_case_visible: true

```

My next PR already includes extra comments and renames the variables produced by the combinator so that they are prefixed by "c_", which should help.